src/codeu/chat/client/core/Controller.java

```diff
@@ -38,6 +42,264 @@ public Controller(ConnectionSource source) {
   }
 
   @Override
+  public Integer toggleRemovedBit(Uuid convo, Uuid user){
+    Integer response = 0;
+
+    try(final Connection connection = source.connect()){
+
+      Serializers.INTEGER.write(connection.out(), NetworkCode.TOGGLE_REMOVED_BIT_REQUEST);
+      Uuid.SERIALIZER.write(connection.out(), convo);
+      Uuid.SERIALIZER.write(connection.out(), user);
+
+      if(Serializers.INTEGER.read(connection.in()) == NetworkCode.TOGGLE_REMOVED_BIT_RESPONSE){
+        response = Serializers.INTEGER.read(connection.in());
+      } else {
+        LOG.error("Response from server failed.");
+      }
+    } catch (Exception ex){
+      System.out.println("ERROR: Exception during call on server. Check log for details.");
+      LOG.error(ex, "Exception during call on server.");
+    }
+
+    return response;
+  }
+
+  @Override
+  public Integer toggleMemberBit(Uuid convo, Uuid user, Boolean flag){
+    Integer response = 0;
+
+    try(final Connection connection = source.connect()){
+
+      Serializers.INTEGER.write(connection.out(), NetworkCode.TOGGLE_MEMBER_BIT_REQUEST);
+      Uuid.SERIALIZER.write(connection.out(), convo);
+      Uuid.SERIALIZER.write(connection.out(), user);
+      Serializers.BOOLEAN.write(connection.out(), flag);
+
+      if(Serializers.INTEGER.read(connection.in()) == NetworkCode.TOGGLE_MEMBER_BIT_RESPONSE){
+        response = Serializers.INTEGER.read(connection.in());
+      } else {
+        LOG.error("Response from server failed.");
+      }
+    } catch (Exception ex){
+      System.out.println("ERROR: Exception during call on server. Check log for details.");
+      LOG.error(ex, "Exception during call on server.");
+    }
+
+    return response;
+  }
+
+  @Override
+  public Integer toggleOwnerBit(Uuid convo, Uuid user, Boolean flag){
+    Integer response = 0;
+
+    try(final Connection connection = source.connect()){
+
+      Serializers.INTEGER.write(connection.out(), NetworkCode.TOGGLE_OWNER_BIT_REQUEST);
+      Uuid.SERIALIZER.write(connection.out(), convo);
+      Uuid.SERIALIZER.write(connection.out(), user);
+      Serializers.BOOLEAN.write(connection.out(), flag);
+
+      if(Serializers.INTEGER.read(connection.in()) == NetworkCode.TOGGLE_OWNER_BIT_RESPONSE){
+        response = Serializers.INTEGER.read(connection.in());
+      } else {
+        LOG.error("Response from server failed.");
+      }
+    } catch (Exception ex){
+      System.out.println("ERROR: Exception during call on server. Check log for details.");
+      LOG.error(ex, "Exception during call on server.");
+    }
+
+    return response;
+  }
+
+  @Override
+  public Integer toggleCreatorBit(Uuid convo, Uuid user, Boolean flag){
```

Comment:
The above three methods are very similar, can you condense the code to be something like:

```
@Override
public Integer toggleCreatorBit(Uuid convo, Uuid user, Boolean flag) {
  return toggleAccessControlBit(convo, user, flag, NetworkCode.TOGGLE_CREATOR_BIT_REQUEST, NetworkCode.TOGGLE_CREATOR_BIT_RESPONSE);
}

private Integer toggleAccessControlBit(Uuid convo, Uuid user, Boolean flag, Integer requestBit, Integer responseBit) {
   ...
  Serializers.INTEGER.write(connection.out(), requestBit);
  ...
  if(Serializers.INTEGER.read(connection.in()) == responseBit) {  ... }
}
```

and reuse that method for all three?